src/test/java/com/gravity9/mongocdc/ChangeStreamTest.java

```diff
@@ -240,6 +240,49 @@ void shouldGracefullyStopAllRegisteredListeners() throws Exception {
         assertDoesNotThrow(manager::stop);
     }
 
+    @Test
+    void givenConfigurationWithUserDefinedKey_eachListenerShouldReceiveSeparateEvents() throws Exception {
+        var configWithUserDefinedKey = MongoConfig.builder()
+                .connectionUri(getConnectionUri())
+                .databaseName(getDatabaseName())
+                .collectionName(getTestCollectionName())
+                .keyName("testId")
```

Comment:
Please try also with non-ID field (some enum string or date without time).

Reply:
It might work on dates, but without time, it will probably all be moved to one partition. It won't work on Strings. I added information to the readme that only ObjectId is supported. Should we support other types?